pkg/util/util.go

```diff
@@ -90,8 +91,8 @@ func ParseEndpoint(endpoint string) (string, string, error) {
 	return scheme, addr, nil
 }
 
-func roundUpSize(volumeSizeBytes int64, allocationUnitBytes int64) int64 {
-	return (volumeSizeBytes + allocationUnitBytes - 1) / allocationUnitBytes
+func roundUpSize(volumeSizeBytes int64, allocationUnitBytes int64) int32 {
+	return int32((volumeSizeBytes + allocationUnitBytes - 1) / allocationUnitBytes)
```

Comment:
I think we can silently overflow with large inputs here.

Casting in several different places is rather confusing and increases the likelihood of bugs. If the expected output from this function is `int32`, can we make sure that the inputs are also `int32`?

Reply:
ack - Thanks for letting us know about that pain point. I generally agree that we should make the minimal set of changes for the migration and not blow up the scope of this PR; maintainers of this repo ought to be aware of this and help with cleaning up the casting situation to prevent regressions in the future.

TLDR - CSI spec expects and returns values in `int64`, for example:
```
int64 volume_capacity_bytes = 2;
```

but the SDK's expected input/output might be of type`int32` for some attributes.

In other words, we need to work towards a state where minimal casting is done - preferably, all inputs are in `int32` where needed, then a single cast to `int64` is performed in a centralized place. That should be safe because `int32` is _implicitly_  convertible to `int64`- every 32 bit int can be expressed as a 64 bit int, so there is no risk of silent overflows, etc.

cc: @jacobwolfaws

Reply:
Makes sense, thanks for the extra detail!
Just thinking, we could leave everything in `RoundUpVolumeSize` in int64, there's only two uses of that function (https://github.com/kubernetes-sigs/aws-fsx-csi-driver/blob/master/pkg/driver/controller.go#L213 and https://github.com/kubernetes-sigs/aws-fsx-csi-driver/blob/master/pkg/driver/controller.go#L395). I could change it there such that they cast to int32 only in those locations, and even check if the number to cast doesn't exceed `Math.MaxInt32`, with an error if it is?